its/src/test/java/com/sonar/it/scanner/msbuild/ScannerMSBuildTest.java

```diff
@@ -816,6 +816,61 @@ public void testIgnoreIssuesDoesNotRemoveSourceGenerator() throws IOException {
     assertThat(filter(issues, ROSLYN_RULES_PREFIX)).isEmpty();
   }
 
+  @Test
+  public void whenEachProjectIsOnDifferentDrives_AnalysisFails() throws IOException {
+    Path projectDir = TestUtils.projectDir(temp, "TwoDrivesTwoProjects");
+
+    int setupStatus = CommandExecutor.create().execute(
+      Command.create("setup.bat")
+        .addArguments(projectDir.resolve( "DriveZ").toAbsolutePath().toString())
+        .setDirectory(projectDir.toFile()),
+      60 * 1000);
+    assertThat(setupStatus).isZero();
+
+    BuildResult buildResult = runAnalysisWithoutProjectBasedDir(projectDir);
+
+    assertThat(buildResult.isSuccess()).isFalse();
+    assertThat(buildResult.getLogs()).contains("Generation of the sonar-properties file failed. Unable to complete the analysis.");
+
+    int cleanupStatus = CommandExecutor.create().execute(Command.create("cleanup.bat").setDirectory(projectDir.toFile()), 60 * 1000);
+    assertThat(cleanupStatus).isZero();
+  }
+
+  @Test
+  public void whenMinorityOfProjectsIsOnDifferentDrives_AnalysisSucceeds() throws IOException {
```

Comment:
Because the semantic of the "win" scenario is defined like this. And that leads to the success
```suggestion
  public void whenMajorityOfProjectsIsOnSameDrive_AnalysisSucceeds() throws IOException {
```